tf2onnx/function/matrixbandpart.py

```diff
@@ -0,0 +1,90 @@
+    const_zero = ctx.make_const(name=node_name, np_val=np.array([0]).astype(np.int32))
+    first_col_or_row = ctx.make_node(op_type="Gather", inputs=[node.input[0], const_zero.output[0]],
+                                     attr={"axis": axis})
+    nodes.append(first_col_or_row)
+    first_col_or_row_casted = ctx.make_node(op_type="Cast", inputs=first_col_or_row.output,
+                                            attr={"to": onnx_pb.TensorProto.BOOL})
+    nodes.append(first_col_or_row_casted)
+    # line means one col or one row
+    zero_line = ctx.make_node(op_type="Xor", inputs=first_col_or_row_casted.output*2)
+    nodes.append(zero_line)
+    one_line = ctx.make_node(op_type="Not", inputs=zero_line.output)
+    nodes.append(one_line)
+
+    # 2: "loop" to generate mask matrix: generate col or row of matrix one by one
+    body_ins = [utils.make_onnx_inputs_outputs("trip", onnx_pb.TensorProto.INT64, []),
+                utils.make_onnx_inputs_outputs("cond", onnx_pb.TensorProto.BOOL, []),
+                utils.make_onnx_inputs_outputs("line", onnx_pb.TensorProto.BOOL, [-1, -1])]
+    body_outs = [utils.make_onnx_inputs_outputs("cond_out", onnx_pb.TensorProto.BOOL, []),
+                 utils.make_onnx_inputs_outputs("line_out", onnx_pb.TensorProto.BOOL, [-1, -1]),
+                 utils.make_onnx_inputs_outputs("res", onnx_pb.TensorProto.BOOL, [-1, -1])]
+    # body graph
+    node_name = utils.make_name("const_zero_bool")
+    const_zero_bool = ctx.make_const(name=node_name, np_val=np.array([[0]]).astype(np.bool))
+    slice_node = ctx.make_node(op_type="Slice", inputs=["line"],
+                               attr={"axes": [counter_axis], "starts": [0], "ends": [-1]})
+    new_line = ctx.make_node(op_type="Concat", inputs=[const_zero_bool.output[0], slice_node.output[0]],
+                             outputs=["line_out"], attr={"axis": counter_axis})
+    body_nodes = list([slice_node.op, new_line.op])
+    body_nodes.append(utils.make_onnx_identity("cond", "cond_out"))
+    body_nodes.append(utils.make_onnx_identity("line", "res"))
+    body_graph = helper.make_graph(body_nodes, utils.make_name("MatrixBandPart_body"), body_ins, body_outs)
+    # initial value of body vars
+    shape = ctx.make_node(op_type="Shape", inputs=[node.input[0]])  # dtype of result is int64
+    nodes.append(shape)
+    node_name = utils.make_name("line_num_index")
+    col_or_row_num_index = ctx.make_const(name=node_name, np_val=np.array(axis).astype(np.int32))
+    line_num = ctx.make_node(op_type="Gather", inputs=[shape.output[0], col_or_row_num_index.output[0]])
+    nodes.append(line_num)
+    trip_cnt = line_num.output[0]
+    node_name = utils.make_name("true")
+    cond = ctx.make_const(name=node_name, np_val=np.array(1).astype(np.bool)).output[0]
+    col_init = one_line.output[0]
+
+    loop_node = ctx.make_node(op_type="Loop", inputs=[trip_cnt, cond, col_init], output_count=2,
+                              attr={"body": body_graph})
```

Comment:
simplily the code using this? 

inputs=[line_num.output[0], cond.output[0], one_line.output[0]]

Reply:
maybe the readability of original code is better since we can know the meaning of data is initializer?